src/tools/perf/lib/ucp_tests.cc

```diff
@@ -432,13 +474,18 @@ class ucp_perf_test_runner {
                 ucx_perf_update(&m_perf, 1, length);
                 ++sn;
             }
+
+            wait_last_iter(recv_buffer, recv_length);
         } else if (my_index == 1) {
+            reserve_last_iter();
             UCX_PERF_TEST_FOREACH(&m_perf) {
                 send(ep, send_buffer, send_length, send_datatype, sn,
                      remote_addr, rkey);
                 ucx_perf_update(&m_perf, 1, length);
                 ++sn;
             }
+            send_last_iter(ep, send_buffer, send_length, send_datatype,
```

Comment:
fence before

Reply:
unhandled

Reply:
@yosefe do you mean `ucp_worker_fence()`?
I don't think that they are needed, since we UCP blocking UCP PUT

Reply:
discussed offline with @yosefe, done